api/v1/cluster_webhook.go

```diff
@@ -1858,6 +1858,17 @@ func (r *Cluster) validateManagedRoles() field.ErrorList {
 					role.Name,
 					"This role is reserved for operator use"))
 		}
+
+		if role.PasswordSecret != nil && role.ValidUntil != "" {
+			if _, err := utils.ParseTargetTime(nil, role.ValidUntil); err != nil {
```

Comment:
if we kept the validUntil timestamp as `*metav1.Time`, we'd get parsing / validation for free